src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -403,7 +402,8 @@ void GcsResourceManager::GetResourceUsageBatchForBroadcast(
   auto beg = resources_buffer_.begin();
   auto ptr = beg;
   for (size_t cnt = buffer.batch().size();
-       cnt < max_broadcasting_batch_size_ && cnt < resources_buffer_.size();
+       buffer->ByteSizeLong() < max_broadcasting_body_size_ &&
```

Comment:
Please avoid using ByteSizeLong. From my memory, it's pretty slow and will crash gcs when there are a lot of nodes.